databroker/experimental/server_ext.py

```diff
@@ -0,0 +1,499 @@
+import base64
+import collections.abc
+import json
+import os
+import uuid
+from pathlib import Path
+from typing import Dict, List
+
+import dask.array
+import h5py
+import numpy
+import pandas
+import pydantic
+import pymongo
+
+from fastapi import APIRouter, HTTPException, Request, Security
+from tiled.adapters.array import ArrayAdapter
+from tiled.adapters.dataframe import DataFrameAdapter
+from tiled.adapters.utils import IndexersMixin, tree_repr
+from tiled.server.core import json_or_msgpack
+from tiled.server.dependencies import entry
+from tiled.query_registration import QueryTranslationRegistry
+
+from tiled.structures.core import StructureFamily
+from tiled.structures.dataframe import deserialize_arrow, serialize_arrow
+
+from tiled.server.pydantic_array import ArrayStructure
+from tiled.server.pydantic_dataframe import DataFrameStructure
+from tiled.utils import UNCHANGED
+
+from typing import Union
+
+from .schemas import Document
+from .queries import RawMongo
+
+from sys import platform
+
+
+class PostMetadataRequest(pydantic.BaseModel):
+    structure_family: StructureFamily
+    structure: Union[ArrayStructure, DataFrameStructure]
+    metadata: Dict
+    specs: List[str]
+    mimetype: str
```

Comment:
After my recent revisions, the client does not include a `mimetype` in the body of its request. The `mimetype` that the data is stored in is up to the server, not the client.